Test/FMI3/fmu_dummy/fmu3_model.c

```diff
@@ -217,53 +238,79 @@ fmi3Status fmi_get_binary(fmi3Instance instance, const fmi3ValueReference vr[],
         return fmi3Fatal;
     }
 
-    for (k = 0; k < nvr; k++) {
-        value[k] = inst->binaries[vr[k]];
+    for (k = 0; k < nValueReferences; k++) {
+        values[k] = inst->binaries[valueReferences[k]];
         sizes[k] = inst->binaries_sz[k];
     }
     return fmi3OK;
 }
 
-/* TODO: nValues is just used for verification, i.e. to check that sum(lenght(vr[i])) == nValues;
- * currently it seems it's not decided whether this parameter will remain, so not implementing anything
- * for now
- * https://github.com/modelica/fmi-standard/issues/512
+/* TODO: nValues is just used for verification, i.e. to check that sum(lenght(valueReferences[i])) == nValues;
+ *
  */
-fmi3Status fmi_set_float64(fmi3Instance instance, const fmi3ValueReference vr[], size_t nvr, const fmi3Float64 value[], size_t nValues)
+fmi3Status fmi_set_float64(fmi3Instance instance, const fmi3ValueReference valueReferences[],
+    size_t nValueReferences, const fmi3Float64 values[], size_t nValues)
 {
     instance_ptr_t inst = instance;
     if (inst == NULL) {
         return fmi3Fatal;
     } else {
+        //if (sizeof(valueReferences)/sizeof(valueReferences[0]) != nValueReferences) {
+        //    inst->cb.logMessage(instance, fmi3Fatal, "FATAL", "Mismatch in length between array of values references and nValues");
+        //    return fmi3Fatal;
+        //}
         size_t k;
-        for (k = 0; k < nvr; k++) {
-            fmi3ValueReference cvr = vr[k];
-            if (cvr < N_STATES) {
-                inst->states[cvr] = value[k];
+        for (k = 0; k < nValueReferences; k++) {
+            fmi3ValueReference currentValueReference = valueReferences[k];
+            if (currentValueReference < N_STATES) {
+                inst->states[currentValueReference] = values[k];
             }
-            else if(cvr == 4) {
-                inst->cb.logMessage(instance, fmi3Warning, "WARNING", "Cannot set acceleration value (calculated)");
+            else if(currentValueReference == 4) {
+                inst->cb.logMessage(instance, fmi3Warning, "WARNING", "Cannot set acceleration values (calculated)");
                 return fmi3Error;
             }
+            else if(currentValueReference == 12) {
+                // For the dummy FMU we are testing an array of 4 elements.
+                //memcpy(&inst->reals[currentValueReference], &values[k], 4 * sizeof(values[0]));
+                for(int i = 0; i < 4; i++) {
+                    inst->reals[currentValueReference + i] = values[i];
```

Comment:
The "reals" array here is used by the simulation tests. For testing `get/set` `float/int` I was not able to do the discussed testing strategy without causing all the simulation tests to fail. I thought it was easier simply to do this way even though it does not look that appealing. I am open to change it, let me know what you think.

Reply:
Maybe it's enough to test the int64 array otherwise? I don't think we gain much from the dummy float64 array?

Reply:
What I meant is that we already do set and get on
```
  <Float64 name="array_of_states" valueReference="12" causality="output" description="Variable testing of arrays. Outputs states and state derivatives: {{h, h'}, {v, v'}} (where h' = v)" start="1.0 4.0 4.0 -9.81">
    <Dimension start="2"/>
    <Dimension start="2"/>
  </Float64>
```
So apparently that seems to already have been tested. I.e. Float64 + array. I think it's more confusing to test the same thing twice. Or is there any difference?

Separate topic: You should add the dummy_int64_array to the XML. Not because we really need it for this test, but to avoid future confusion.